tests/test_server_proto.py

```diff
@@ -1888,6 +1892,138 @@ def on_log(con, msg):
                 result, "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
 
 
+class TestServerProtoMigration(tb.QueryTestCase):
+
+    ISOLATED_METHODS = False
+
+    async def test_server_proto_mig_01(self):
+        # Replicating the "test_edgeql_tutorial" test that might
+        # disappear at some point. That test was the only one that
+        # uncovered a regression in how server schema state is
+        # handled, so we need to keep some form of it.
+
+        typename = f'test_{uuid.uuid4().hex}'
+
+        await self.con.execute(f'''
+            START TRANSACTION;
+            CREATE MIGRATION def TO {{
+                module default {{
+                    type {typename} {{
+                        required property foo -> str;
+                    }}
+                }}
+            }};
+            COMMIT MIGRATION def;
+            COMMIT;
+
+            INSERT {typename} {{
+                foo := '123'
+            }};
+        ''')
+
+        await self.assert_query_result(
+            f'SELECT {typename}.foo',
+            ['123']
+        )
+
+
+class TestServerProtoDdlPropagation(tb.QueryTestCase):
+
+    ISOLATED_METHODS = False
+
+    async def test_server_proto_ddlprop_01(self):
+        conargs = self.get_connect_args()
+
+        settings = self.con.get_settings()
+        pgaddr = settings.get('pgaddr')
+        if pgaddr is None:
+            raise RuntimeError('test requires devmode')
+        pgaddr = json.loads(pgaddr)
+        pgdsn = (
+            f'postgres:///?user={pgaddr["user"]}&port={pgaddr["port"]}'
+            f'&host={pgaddr["host"]}'
+        )
+
+        await self.con.execute('''
+            CREATE TYPE Test {
+                CREATE PROPERTY foo -> int16;
+            };
+
+            INSERT Test { foo := 123 };
+        ''')
+
+        self.assertEqual(
+            await self.con.fetchone('SELECT Test.foo LIMIT 1'),
+            123
+        )
+
+        with tempfile.TemporaryDirectory() as tmp:
```

Comment:
Might be worth factoring this out into `cluster.py` or `testbase` once we start adding other tests like this.

Reply:
Yeah, as soon as we need a second test like this I'll add a reusable harness.